Text/Lucius.hs

```diff
@@ -86,39 +105,56 @@ whiteSpace1 :: Parser ()
 whiteSpace1 =
     ((oneOf " \t\n\r" >> return ()) <|> (parseComment >> return ()))
 
-parseBlock :: Parser (Block Unresolved)
-parseBlock = do
+parseBlock :: Order -> Parser (Block 'Unresolved)
+parseBlock order = do
     sel <- parseSelector
     _ <- char '{'
     whiteSpace
-    pairsBlocks <- parsePairsBlocks id
-    let (pairs, blocks, mixins) = partitionPBs pairsBlocks
+    pairsBlocks <- parsePairsBlocks order id
+    let (attrs, blocks, mixins) = case order of
+          Unordered -> partitionPBs  pairsBlocks
+          Ordered   -> partitionPBs' pairsBlocks
     whiteSpace
-    return $ Block sel pairs (map detectAmp blocks) mixins
+    return $ BlockUnresolved sel attrs (map detectAmp blocks) mixins
 
 -- | Looks for an & at the beginning of a selector and, if present, indicates
 -- that we should not have a leading space. Otherwise, we should have the
 -- leading space.
-detectAmp :: Block Unresolved -> (Bool, Block Unresolved)
-detectAmp (Block (sel) b c d) =
-    (hls, Block sel' b c d)
+detectAmp :: Block 'Unresolved -> (Bool, Block 'Unresolved)
+detectAmp (BlockUnresolved (sel) b c d) =
+    (hls, BlockUnresolved sel' b c d)
   where
     (hls, sel') =
         case sel of
             (ContentRaw "&":rest):others -> (False, rest : others)
             (ContentRaw ('&':s):rest):others -> (False, (ContentRaw s : rest) : others)
             _ -> (True, sel)
 
-partitionPBs :: [PairBlock] -> ([Attr Unresolved], [Block Unresolved], [Deref])
+-- | This function implementation does not keep the order
+-- of attributes and mixins and is used in legacy parsers:
+-- 'cassius', 'cassiusFile', 'lucius', 'luciusFile' etc.
+-- To maintain attributes and mixins ordering, use 'partitionPBs'.
```

Comment:
Missing the `'` in the function name at the end.

Reply:
Not relevant in this edition.